src/qt/paymentserver.cpp

```diff
@@ -643,8 +643,9 @@ void PaymentServer::fetchPaymentACK(CWallet* wallet, const SendCoinsRecipient& r
         // use for change. Despite an actual payment and not change, this is a close match:
         // it's the output type we use subject to privacy issues, but not restricted by what
         // other software supports.
-        wallet->LearnRelatedScripts(newKey, g_change_type);
-        CTxDestination dest = GetDestinationForKey(newKey, g_change_type);
+        OutputType change_type = g_change_type ? g_change_type : g_address_type;
```

Comment:
Shouldn't this compare explicitly? ie `change_type = g_change_type != OUTPUT_TYPE_NONE ? g_change_type : g_address_type`

Reply:
I wanted to keep it short for readability and leverage that OUTPUT_TYPE_NONE is 0. I can change that if you think it's unsafe.

Reply:
I think it should be changed (in the future). The whole enum should be a c++11 enum class to be type safe. Though, that will touch all the code, so maybe too late for 0.16...